[gettext] Build error on x64-freebsd

Package: gettext[core,tools]:x64-freebsd@0.22.4#1

**Host Environment**

- Host: x64-freebsd
- Compiler: Clang 17.0.6
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: c268e9fc3 2024-01-16 (2 hours ago)

**To Reproduce**

`vcpkg install gettext[core,tools]`
**Failure logs**

```
-- Using cached gettext-0.22.4.tar.gz.
-- Cleaning sources at /usr/home/alonso/projects/vcpkg/buildtrees/gettext/src/gettext-0-cd96ae6806.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /usr/home/alonso/projects/vcpkg/downloads/gettext-0.22.4.tar.gz
-- Applying patch assume-modern-darwin.patch
-- Applying patch uwp.patch
-- Applying patch rel_path.patch
-- Applying patch subdirs.patch
-- Applying patch parallel-gettext-tools.patch
-- Applying patch config-step-order.patch
-- Using source at /usr/home/alonso/projects/vcpkg/buildtrees/gettext/src/gettext-0-cd96ae6806.clean
-- Found external ninja('1.11.1').
-- Getting CMake variables for x64-freebsd-rel
-- Configuring x64-freebsd-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /usr/local/bin/bash -c "V=1 ./../src/gettext-0-cd96ae6806.clean/configure  \"--enable-relocatable\" \"--enable-c++\" \"--disable-acl\" \"--disable-csharp\" \"--disable-curses\" \"--disable-java\" \"--disable-openmp\" \"--disable-dependency-tracking\" \"--with-included-glib\" \"--with-included-libxml\" \"--with-included-libunistring\" \"--with-installed-libtextstyle=no\" \"--without-cvs\" \"--without-emacs\" \"--without-git\" \"--without-libcurses-prefix\" \"--without-libncurses-prefix\" \"--without-libtermcap-prefix\" \"--without-libxcurses-prefix\" \"INTLBISON=bison\" \"TOOLS_BISON=bison\" \"--with-included-gettext=no\" \"--disable-silent-rules\" \"--verbose\" \"--enable-shared\" \"--disable-static\" \"--cache-file=/usr/home/alonso/projects/vcpkg/buildtrees/gettext/config.cache-x64-freebsd-rel.log\" \"--prefix=/usr/home/alonso/projects/vcpkg/installed/x64-freebsd\" \"--bindir=\\${prefix}/tools/gettext/bin\" \"--sbindir=\\${prefix}/tools/gettext/sbin\" \"--libdir=\\${prefix}/lib\" \"--mandir=\\${prefix}/share/gettext\" \"--docdir=\\${prefix}/share/gettext\" \"--datarootdir=\\${prefix}/share/gettext\""
    Working Directory: /usr/home/alonso/projects/vcpkg/buildtrees/gettext/x64-freebsd-rel
    Error code: 2
    See logs for more information:
      /usr/home/alonso/projects/vcpkg/buildtrees/gettext/config-x64-freebsd-rel-config.log
      /usr/home/alonso/projects/vcpkg/buildtrees/gettext/config-x64-freebsd-rel-out.log
      /usr/home/alonso/projects/vcpkg/buildtrees/gettext/config-x64-freebsd-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:832 (vcpkg_execute_required_process)
  ports/gettext/portfile.cmake:142 (vcpkg_configure_make)
  scripts/ports.cmake:170 (include)



```
<details><summary>/usr/home/alonso/projects/vcpkg/buildtrees/gettext/config-x64-freebsd-rel-err.log</summary>

```
./../src/gettext-0-cd96ae6806.clean/configure: ${VCPKG_GETTEXT_SUBDIRS/...}: Bad substitution
```
</details>

<details><summary>/usr/home/alonso/projects/vcpkg/buildtrees/gettext/config-x64-freebsd-rel-out.log</summary>

```
configure: creating cache /usr/home/alonso/projects/vcpkg/buildtrees/gettext/config.cache-x64-freebsd-rel.log
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/local/bin/gmkdir -p
checking for gawk... no
checking for mawk... no
checking for nawk... nawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '1001' is supported by ustar format... yes
checking whether GID '20' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
```
</details>

<details><summary>/usr/home/alonso/projects/vcpkg/buildtrees/gettext/config-x64-freebsd-rel-config.log</summary>

```
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by gettext configure 0.22.4, which was
generated by GNU Autoconf 2.71.  Invocation command line was

  $ ./../src/gettext-0-cd96ae6806.clean/configure --enable-relocatable --enable-c++ --disable-acl --disable-csharp --disable-curses --disable-java --disable-openmp --disable-dependency-tracking --with-included-glib --with-included-libxml --with-included-libunistring --with-installed-libtextstyle=no --without-cvs --without-emacs --without-git --without-libcurses-prefix --without-libncurses-prefix --without-libtermcap-prefix --without-libxcurses-prefix INTLBISON=bison TOOLS_BISON=bison --with-included-gettext=no --disable-silent-rules --verbose --enable-shared --disable-static --cache-file=/usr/home/alonso/projects/vcpkg/buildtrees/gettext/config.cache-x64-freebsd-rel.log --prefix=/usr/home/alonso/projects/vcpkg/installed/x64-freebsd '--bindir=${prefix}/tools/gettext/bin' '--sbindir=${prefix}/tools/gettext/sbin' '--libdir=${prefix}/lib' '--mandir=${prefix}/share/gettext' '--docdir=${prefix}/share/gettext' '--datarootdir=${prefix}/share/gettext'

## --------- ##
## Platform. ##
## --------- ##

hostname = sunstream.divine.net
uname -m = amd64
uname -r = 13.2-STABLE
uname -s = FreeBSD
uname -v = FreeBSD 13.2-STABLE stable/13-n257175-c8a0dbcaff3f GENERIC

/usr/bin/uname -p = amd64
/bin/uname -X     = unknown

/bin/arch              = unknown
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = unknown
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /usr/local/bin/
PATH: /usr/local/sbin/
PATH: /usr/bin/
PATH: /usr/sbin/
PATH: /bin/
PATH: /sbin/
PATH: /usr/local/bin/
PATH: /usr/local/bin/
PATH: /usr/local/bin/


## ----------- ##
## Core tests. ##
## ----------- ##

configure:1888: creating cache /usr/home/alonso/projects/vcpkg/buildtrees/gettext/config.cache-x64-freebsd-rel.log
configure:1907: looking for aux files: config.guess config.sub missing install-sh
configure:1920:  trying ./../src/gettext-0-cd96ae6806.clean/build-aux/
configure:1949:   ./../src/gettext-0-cd96ae6806.clean/build-aux/config.guess found
configure:1949:   ./../src/gettext-0-cd96ae6806.clean/build-aux/config.sub found
configure:1949:   ./../src/gettext-0-cd96ae6806.clean/build-aux/missing found
configure:1931:   ./../src/gettext-0-cd96ae6806.clean/build-aux/install-sh found
configure:2080: checking for a BSD-compatible install
configure:2153: result: /usr/local/bin/ginstall -c
configure:2164: checking whether build environment is sane
configure:2219: result: yes
configure:2378: checking for a race-free mkdir -p
configure:2422: result: /usr/local/bin/gmkdir -p
configure:2429: checking for gawk
configure:2464: result: no
configure:2429: checking for mawk
configure:2464: result: no
configure:2429: checking for nawk
configure:2450: found /usr/bin/nawk
configure:2461: result: nawk
configure:2472: checking whether make sets $(MAKE)
configure:2495: result: yes
configure:2525: checking whether make supports nested variables
configure:2543: result: yes
configure:2628: checking whether UID '1001' is supported by ustar format
configure:2631: result: yes
configure:2638: checking whether GID '20' is supported by ustar format
configure:2641: result: yes
configure:2649: checking how to create a ustar tar archive
configure:2660: tar --version
bsdtar 3.6.2 - libarchive 3.6.2 zlib/1.3 liblzma/5.4.5 bz2lib/1.0.8 libzstd/1.4.8 
configure:2663: $? = 0
configure:2703: tardir=conftest.dir && eval tar --format=ustar -chf - "$tardir" >conftest.tar
configure:2706: $? = 0
configure:2710: tar -xf - <conftest.tar
configure:2713: $? = 0
configure:2715: cat conftest.dir/file
GrepMe
configure:2718: $? = 0
configure:2732: result: gnutar

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_env_CC_set=set
ac_cv_env_CC_value=/usr/bin/cc
ac_cv_env_CFLAGS_set=set
ac_cv_env_CFLAGS_value='-fPIC -O3'
ac_cv_env_CPPFLAGS_set=set
ac_cv_env_CPPFLAGS_value=-DNDEBUG
ac_cv_env_CPP_set=''
ac_cv_env_CPP_value=''
ac_cv_env_CXXCPP_set=''
ac_cv_env_CXXCPP_value=''
ac_cv_env_CXXFLAGS_set=set
ac_cv_env_CXXFLAGS_value='-fPIC -O3'
ac_cv_env_CXX_set=set
ac_cv_env_CXX_value=/usr/bin/c++
ac_cv_env_EMACSLOADPATH_set=''
ac_cv_env_EMACSLOADPATH_value=''
ac_cv_env_EMACS_set=''
ac_cv_env_EMACS_value=''
ac_cv_env_LDFLAGS_set=set
ac_cv_env_LDFLAGS_value=-L/usr/home/alonso/projects/vcpkg/installed/x64-freebsd/lib
ac_cv_env_LIBS_set=''
ac_cv_env_LIBS_value=''
ac_cv_env_LT_SYS_LIBRARY_PATH_set=''
ac_cv_env_LT_SYS_LIBRARY_PATH_value=''
ac_cv_env_YACC_set=''
ac_cv_env_YACC_value=''
ac_cv_env_YFLAGS_set=''
ac_cv_env_YFLAGS_value=''
ac_cv_env_build_alias_set=''
ac_cv_env_build_alias_value=''
ac_cv_env_host_alias_set=''
ac_cv_env_host_alias_value=''
ac_cv_env_target_alias_set=''
ac_cv_env_target_alias_value=''
ac_cv_path_install='/usr/local/bin/ginstall -c'
ac_cv_path_mkdir=/usr/local/bin/gmkdir
ac_cv_prog_AWK=nawk
ac_cv_prog_make_make_set=yes
am_cv_make_support_nested_variables=yes
am_cv_prog_tar_ustar=gnutar

## ----------------- ##
## Output variables. ##
## ----------------- ##

ACLOCAL='${SHELL} '\''/usr/home/alonso/projects/vcpkg/buildtrees/gettext/src/gettext-0-cd96ae6806.clean/build-aux/missing'\'' aclocal-1.16'
AMTAR='$${TAR-tar}'
AM_BACKSLASH='\'
AM_DEFAULT_V='$(AM_DEFAULT_VERBOSITY)'
AM_DEFAULT_VERBOSITY='1'
AM_V='$(V)'
AUTOCONF='${SHELL} '\''/usr/home/alonso/projects/vcpkg/buildtrees/gettext/src/gettext-0-cd96ae6806.clean/build-aux/missing'\'' autoconf'
AUTOHEADER='${SHELL} '\''/usr/home/alonso/projects/vcpkg/buildtrees/gettext/src/gettext-0-cd96ae6806.clean/build-aux/missing'\'' autoheader'
AUTOMAKE='${SHELL} '\''/usr/home/alonso/projects/vcpkg/buildtrees/gettext/src/gettext-0-cd96ae6806.clean/build-aux/missing'\'' automake-1.16'
AWK='nawk'
CC='/usr/bin/cc'
CFLAGS='-fPIC -O3'
CPP=''
CPPFLAGS='-DNDEBUG'
CSCOPE='cscope'
CTAGS='ctags'
CXX='/usr/bin/c++'
CXXCPP=''
CXXFLAGS='-fPIC -O3'
CYGPATH_W='echo'
DEFS=''
ECHO_C=''
ECHO_N='-n'
ECHO_T=''
EMACS=''
EMACSLOADPATH=''
ETAGS='etags'
INSTALL_DATA='${INSTALL} -m 644'
INSTALL_PROGRAM='${INSTALL}'
INSTALL_SCRIPT='${INSTALL}'
INSTALL_STRIP_PROGRAM='$(install_sh) -c -s'
LDFLAGS='-L/usr/home/alonso/projects/vcpkg/installed/x64-freebsd/lib'
LIBOBJS=''
LIBS=''
LTLIBOBJS=''
LT_SYS_LIBRARY_PATH=''
MAKEINFO='${SHELL} '\''/usr/home/alonso/projects/vcpkg/buildtrees/gettext/src/gettext-0-cd96ae6806.clean/build-aux/missing'\'' makeinfo'
MKDIR_P='/usr/local/bin/gmkdir -p'
PACKAGE='gettext'
PACKAGE_BUGREPORT='bug-gettext@gnu.org'
PACKAGE_NAME='gettext'
PACKAGE_STRING='gettext 0.22.4'
PACKAGE_TARNAME='gettext'
PACKAGE_URL=''
PACKAGE_VERSION='0.22.4'
PATH_SEPARATOR=':'
SET_MAKE=''
SHELL='/bin/sh'
STRIP='/usr/bin/llvm-strip'
VERSION='0.22.4'
YACC=''
YFLAGS=''
am__isrc=' -I$(srcdir)'
am__leading_dot='.'
am__tar='${AMTAR} chf - --format=ustar --owner=root --group=root "$$tardir"'
am__untar='tar -xf -'
bindir='${prefix}/tools/gettext/bin'
build=''
build_alias=''
build_cpu=''
build_os=''
build_vendor=''
datadir='${datarootdir}'
datarootdir='${prefix}/share/gettext'
docdir='${prefix}/share/gettext'
dvidir='${docdir}'
exec_prefix='NONE'
host=''
host_alias=''
host_cpu=''
host_os=''
host_vendor=''
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
install_sh='${SHELL} /usr/home/alonso/projects/vcpkg/buildtrees/gettext/src/gettext-0-cd96ae6806.clean/build-aux/install-sh'
libdir='${prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${prefix}/share/gettext'
mkdir_p='$(MKDIR_P)'
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='/usr/home/alonso/projects/vcpkg/installed/x64-freebsd'
program_transform_name='s,x,x,'
psdir='${docdir}'
runstatedir='${localstatedir}/run'
sbindir='${prefix}/tools/gettext/sbin'
sharedstatedir='${prefix}/com'
subdirs=''
sysconfdir='${prefix}/etc'
target_alias=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME "gettext"
#define PACKAGE_TARNAME "gettext"
#define PACKAGE_VERSION "0.22.4"
#define PACKAGE_STRING "gettext 0.22.4"
#define PACKAGE_BUGREPORT "bug-gettext@gnu.org"
#define PACKAGE_URL ""
#define PACKAGE "gettext"
#define VERSION "0.22.4"

configure: exit 2
```
</details>

--- a/ports/gettext/subdirs.patch
+++ b/ports/gettext/subdirs.patch
@@ -7,7 +7,7 @@ index 904bdf5..e751ffc 100755
  
  
 -subdirs="$subdirs gettext-runtime libtextstyle gettext-tools"
-+subdirs="$subdirs gettext-runtime ${VCPKG_GETTEXT_SUBDIRS/gettext-runtime/}"
++subdirs="$subdirs gettext-runtime $(echo ${VCPKG_GETTEXT_SUBDIRS} | sed 's/gettext-runtime//')"
  
  
  

--- a/ports/gettext/vcpkg.json
+++ b/ports/gettext/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "gettext",
   "version": "0.22.4",
-  "port-version": 1,
+  "port-version": 2,
   "description": "A GNU framework to help produce multi-lingual messages.",
   "homepage": "https://www.gnu.org/software/gettext/",
   "license": "GPL-3.0-only",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2934,7 +2934,7 @@
     },
     "gettext": {
       "baseline": "0.22.4",
-      "port-version": 1
+      "port-version": 2
     },
     "gettext-libintl": {
       "baseline": "0.22.4",

--- a/versions/g-/gettext.json
+++ b/versions/g-/gettext.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d9cc9b8256d9653fe2d484ecf268f7368666d942",
+      "version": "0.22.4",
+      "port-version": 2
+    },
+    {
       "git-tree": "bd31edd407c0889c864c1e8854be92b602a4e29f",
       "version": "0.22.4",
       "port-version": 1

